iohk/UpdateProposal.hs

```diff
@@ -437,36 +427,36 @@ updateProposalFindInstallers opts env = do
   void $ doCheckConfig params
   echo "*** Finding installers"
   let rev = unGitRevision . cfgDaedalusRevision $ params
-  res <- using $ realFindInstallers rev (configurationKeys env)
+  tempdir <- mktempdir "/tmp" "iohk-ops"
+  res <- liftIO $ realFindInstallers (configurationKeys env) (installerForEnv env) rev (Just tempdir)
   echo "*** Finished. Moving files to work dir"
   res' <- moveInstallersToWorkDir opts res
   writeWikiRecord opts res'
   storeParams opts (UpdateProposalConfig2 params res')
 
+installerForEnv :: Environment -> CIResult -> Bool
+installerForEnv env = matchNet . installerNetwork . ciResultLocalPath
+  where matchNet n = case env of
+          Production -> n == Just InstallerMainnet
+          Staging    -> n == Just InstallerStaging
+          _          -> True
+
 -- | Move installer files from wherever they were found into the work dir.
 moveInstallersToWorkDir :: CommandOptions -> InstallersResults -> Shell InstallersResults
-moveInstallersToWorkDir opts InstallersResults{..} = do
-  bk <- case buildkiteResult of
-    Just res -> do
-      p <- move (bkLocalPath res)
-      pure . Just $ res { bkLocalPath = p }
-    Nothing -> pure Nothing
-  av <- case appveyorResult of
-    Just res -> do
-      p <- move (avLocalPath res)
-      pure . Just $ res { avLocalPath = p }
-    Nothing -> pure Nothing
-  pure $ InstallersResults
-    { buildkiteResult = bk
-    , appveyorResult = av
-    , globalResult = globalResult }
+moveInstallersToWorkDir opts irs = do
+  cis <- mapM moveInstaller (ciResults irs)
+  pure irs { ciResults = cis }
   where
-    move :: Text -> Shell Text
+    moveInstaller :: (Arch, CIResult) -> Shell (Arch, CIResult)
+    moveInstaller (arch, res) = do
+      p <- move (ciResultLocalPath res)
+      pure (arch, res { ciResultLocalPath = p })
+    move :: FilePath -> Shell FilePath
     move src = do
-      let src' = fromText src
-      let dst = installersPath opts (filename src')
-      mv src' dst
-      pure (format fp dst)
+      let dst = installersPath opts (filename src)
+      mv src dst
+      void $ chmod writable dst
```

Comment:
1: this fails to chmod the files because they are in the nix store
2: the installers shouldn't be writable
3: by moving them, nix will no longer know where they are, and garbage-collection can eat the installers, the proper directory should just be passed directly to `realFindInstallers`

Reply:
Thanks, fixed. Also pushed some error message fixes.